test/interactive_tarantool.lua

```diff
@@ -138,11 +168,16 @@ end
 -- * ReadLine 7: x<CR><duplicate x>, extra spaces.
 -- * ReadLine 6: <space><CR>.
 --
--- This function drop duplicates that goes in row, strips <CR> and
--- spaces. Applying of this function to a source command and
--- readline's echoed command allows to compare them for equality.
+-- This function drop duplicates that goes in row, strips <CR>,
+-- spaces, tabs and ANSI color codes. Applying of this function
+-- to a source command and readline's echoed command allows to
+-- compare them for equality.
 local function _prepare_command_for_compare(x)
-    x = x:gsub(' ', ''):gsub(M.CR, '')
+    x = x:gsub(' ', '')
```

Comment:
I believe our goal is not preserving original style but producing patches with good quality.

Reply:
> Please refer to a style-guide rule. If there isn't one, it seems like a matter of taste and can be left as is.

Actually, no, it is not a matter of taste.

Style guide cannot contain all possible rules. If we try to describe all the possible rules to follow and rules that no one should break, then we will be like lawmakers, but this is not our job.

Instead of referring to a certain rule in a style guide, I will appeal to common sense and my understanding of a good programming habits, to respecting expectations of people who will read my code and use these programs.

It could be unexpected when function mutates passed arguments. Function has no any comment so without reading source code you will never know that `x` was updated. Sometimes such behavior in public functions could be a reason of real bugs, see https://github.com/tarantool/crud/issues/192.

Reply:
The bug you referred to is about a different matter. In Lua, when string is passed as an argument, like in `_prepare_command_for_compare()`, it is being copied and local variable is modified, unlike the example in https://github.com/tarantool/crud/issues/192, where a table is passed by reference.